logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/ConversationEventReceiver.kt

```diff
@@ -141,7 +148,41 @@ class ConversationEventReceiverImpl(
             }
     }
 
-    private suspend fun getReadableMessageContent(
+    private suspend fun handleFailedProteusDecryptedMessage(event: Event.Conversation.NewMessage) {
+        with(event) {
+            val message = Message.Regular(
+                id = id,
+                content = MessageContent.FailedDecryption(encryptedExternalContent?.data),
+                conversationId = conversationId,
+                date = timestampIso,
+                senderUserId = senderUserId,
+                senderClientId = senderClientId,
+                status = Message.Status.SENT,
+                editStatus = Message.EditStatus.NotEdited,
+                visibility = Message.Visibility.VISIBLE
+            )
+            processMessage(message)
+        }
+    }
```

Comment:
Would having this be always saving the failed decryption message as forced `editStatus = NotEdited`?

I might be wrong, but aren't then we assuming that and edit will never fail to decrypt?

Reply:
Good question! My understanding is that the message decryption will be always be triggered on newly received messages, so these ones can't be edited messages. Once these are decrypted successfully, they will be persisted unencrypted on our SQLite database and they will never adopt this error message type anymore.

Reply:
To find out if it's an "edit" message, we need to decrypt the content first, so if error happens at this point, we don't know yet what type of message it is, but still we need to store this message, so I think storing it as a not edited FailedDecryption is ok because we don't want any FailedDecryption message type to appear as an edited one.